pkg/reconcile/reconcilers.go

```diff
@@ -185,49 +190,42 @@ func (r *defaultReconciler) ReconcileSchema(ctx context.Context, req ReaperReque
 	}
 }
 
-func (r *defaultReconciler) checkForCassandraDatacenterReadiness(ctx context.Context, req ReaperRequest) error {
+func (r *defaultReconciler) checkForCassandraDatacenterReadiness(ctx context.Context, req ReaperRequest) (bool, error) {
 	reaper := req.Reaper
-	cassandra := *reaper.Spec.ServerConfig.CassandraBackend
-	cassdc := &cassdcapi.CassandraDatacenter{}
+	cassdc, err := r.cassandraDatacenter(ctx, reaper)
+	//if meta.IsNoMatchError(err) {
+	//	return true, nil
+	//}
+	//
+	if err != nil {
+		req.Logger.Error(err, "failed to fetch CassandraDatacenter")
+		return false, err
+	}
 
-	fetchStatus := func() error {
-		cassdcKey := types.NamespacedName{Namespace: reaper.Namespace, Name: cassandra.DatacenterName()}
-		err := r.Client.Get(ctx, cassdcKey, cassdc)
-		if runtime.IsNotRegisteredError(err) {
-			return nil
-		}
-		if err != nil {
-			req.Logger.Error(err, "failed to fetch CassandraDatacenter")
-			return err
+	if value, found := cassdc.Annotations["reaper.cassandra-reaper.io/instance"]; found {
```

Comment:
We actually do not need to check for the `reaper.cassandra-reaper.io/instance` annotation here. The purpose of that annotation is to tell reaper-operator that the CassandraDatacenter should be managed by Reaper for repairs. The backend storage does not have to managed for repairs.